server/py/framework/db/sqldb/db.py

```diff
@@ -1858,7 +1858,11 @@ def _find_artifacts(
             partition_order,
             parent_uri,
         ):
-            query = query.with_hint(ArtifactV2, "USE INDEX idx_project_bi_updated")
+            query = query.with_hint(
+                ArtifactV2,
+                "USE INDEX (idx_project_bi_updated)",
+                dialect_name="mysql",
+            )
```

Comment:
lets try capturing it with a unit test to ensure that "default list (tag=latest, best-iteration=true) will get to here

also, dialect_name=mysql = what's the impact on postgres on that case?